### PR TITLE
implement shaders rewrites and dx9 shader compilation script 

### PR DESCRIPTION
**Pull request by Berthalamew**
- "build dx9 shaders.bat" compiles the .fx files and populates the bytecode into headers
- include xxd and sed from w64devkit
- rewrite weather, fog, and particle shaders for fixed effects
- fix snow size depending on resolution
- include d3d9 debug info in debug builds

### CODE REVIEW

File: xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fog.cpp [hunk 1]
```diff
@@ -1,8 +1,309 @@
...
+	// that is used as a texture in the patchy fog draw code shader
+	WritePointer(Memory::GetAddress(0x2774AE) + 1, rasterizer_dx9_patchy_fog_apply_build_vertex_buffer);
+	WritePointer(Memory::GetAddress(0x2774CF) + 1, rasterizer_dx9_patchy_fog_apply_from_stencil_build_vertex_buffer);
+
+
+	rasterizer_dx9_atmospheric_fog_patch();
+	return;
+}
+
+bool __cdecl rasterizer_dx9_draw_atmospheric_fog(int32 a1)
```
Comment (nukeulater) [inline xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fog.cpp]: this sets up the pipeline, doesn't render anything at all

File: xlive/Blam/Engine/render/weather_definitions.h [hunk 2]
```diff
@@ -1,10 +1,10 @@
 #pragma once
-#include "geometry/geometry_block.h"
 #include "math/color_math.h"
+#include "render/render_weather.h"
 #include "tag_files/tag_reference.h"
 
-#define MAX_PARTICLES_LITE = 4096
 #define MAX_WIND_PRIMITIVES = 128
```
Comment (nukeulater) [inline xlive/Blam/Engine/render/weather_definitions.h]: = 128?

Comment (Berthalamew) [reply xlive/Blam/Engine/render/weather_definitions.h]: fixed

File: xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fog.cpp (hunk 1 above)
Comment (Berthalamew) [reply xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fog.cpp]: function name changed to `rasterizer_dx9_atmospheric_fog_pipeline_setup`

File: xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fullscreen_passes.cpp [hunk 3]
```diff
@@ -11,30 +11,92 @@
...
+    const real32 height = bounds->y1 - bounds->y0;
+
+    output[0].u = (0.5f / width) + location->u;
+    output[0].v = (0.5f / height) + location->v;
+    output[1].u = 0.f;
+    output[1].v = 1.f;
+    return;
+}
 
+void rasterizer_dx9_fullscreen_calculate_texcoords(const real_rectangle2d* bounds, const real_point2d* location, real_point2d* output)
```
Comment (nukeulater) [inline xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fullscreen_passes.cpp]: could probably make the texture index a param here

File: xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_main.cpp [hunk 4]
```diff
@@ -99,10 +107,14 @@ IDirect3DPixelShader9** local_pixel_shaders_get(void)
 void rasterizer_dx9_main_apply_patches(void)
 {
     // debugging, get vertex decls currently set
-    //PatchCall(Memory::GetAddress(0x2220CA), DrawPrimitiveUP_hook_get_vertex_decl);
-    //PatchCall(Memory::GetAddress(0x27D746), DrawPrimitiveUP_hook_get_vertex_decl);
+    PatchCall(Memory::GetAddress(0x2220CA), DrawPrimitiveUP_hook_get_vertex_decl);
```
Comment (nukeulater) [inline xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_main.cpp]: shouldn't be enabled

Comment (Berthalamew) [reply xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_main.cpp]: removed

File: xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fullscreen_passes.cpp (hunk 3 above)
Comment (Berthalamew) [reply xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fullscreen_passes.cpp]: added parameter with default of 0

**Pull request merged**